substrate/frame/identity/src/lib.rs

```diff
@@ -757,8 +783,13 @@ pub mod pallet {
 				return Err(Error::<T>::JudgementGiven.into())
 			};
 
-			let err_amount = T::Currency::unreserve(&sender, fee);
-			debug_assert!(err_amount.is_zero());
+			let released_amount = T::Balances::release(
+				&HoldReason::Judgement.into(),
+				&sender,
+				fee,
+				Precision::Exact,
+			)?;
+			debug_assert!(!released_amount.is_zero());
```

Comment:
Fungible::release is fallible operation in case it fails, contrary to unreserve, I think we can remove those `debug_assert`.